lib/log_parser.rb

```diff
@@ -0,0 +1,17 @@
+class LogParser
+  def initialize(file_path)
+    @file_path = file_path
+  end
+
+  def log_reader
+    if(File.exist?(@file_path))
+      @file = File.open(@file_path, "r")
```

Comment:
```suggestion
      file = File.open(@file_path, "r")
```

@EduardoSCosta costuma ser uma má prática declarar variáveis de instância(com o `@`) fora do initializer. Aqui você pode declarar de forma normal

Para entender um pouco melhor a diferença entre as declarações:
https://www.campuscode.com.br/conteudos/variaveis-e-self-em-ruby